content/blog/2022/2022-09-02-junit5.md

```diff
@@ -0,0 +1,601 @@
+}
+```
+Notice the flexibility we have on the Matchers we pass, there are [many more Matchers](http://hamcrest.org/JavaHamcrest/javadoc/2.2/) we can use on many more data types.
+
+Let's explore those Matchers further more.
+
+### instanceOf
+It tests whether an object is an instance of certain class.
+Say we have a Cat class
+```java
+public class Cat {  
+}
+```
+```java
+public class CatTest {  
+	@Test  
+	public void testCatInstance() {  
+	    Cat cat = new Cat();  
+	  
+	    assertThat(cat, instanceOf(Cat.class));  
+	}
+}
+``` 
+
+### sameInstance
+It verifies if two object references are the same instance.
+```java
+public class CatTest {  
+	@Test  
+	public void testSameCatInstance() {  
+	    Cat cat = new Cat();
+	    assertThat(cat, sameInstance(cat));  
+	}
+}
+```
+
+### hasItems
+It verifies that the given items exist in the collection under test. The existence is checked through the `equals` object method.
+
+```java
+public class CatTest {  
+	@Test  
+	public void testCollectionContaining() {  
+	    List<String> catNames = asList("Phibi", "Monica", "Stilla");  
+	  
+	    assertThat(catNames, hasItems("Monica", "Phibi"));  
+	    assertThat(catNames, not(hasItems("Melih")));  
+	}
+}
+```
+Note the chaining of calls performed with `not(hasItems("Melih"))`, this allows us to mix and match those operators to have complex and readable assertions.
+
+### hasSize
+It checks the size of a given collection against an expected size.
+```java
+public class CatTest {  
+	@Test  
+	public void testCollectionSize() {  
+	    List<String> catNames = asList("Phibi", "Monica");  
+	  
+	    assertThat(catNames, hasSize(2));  
+	}
+}
+``` 
+
+### hasProperty
+This is used to check if an object's property (field) matches a certain condition.
+
+First let's modify our Cat class to have a `name` property with a getter and a constructor.
+
+```java
+public class Cat {  
+    private String name;  
+  
+    public Cat(String name) {  
+        this.name = name;  
+    }  
+  
+    public String getName() {  
+        return name;  
+    }
+ }
+```
+
+```java
+public class CatTest {  
+	@Test  
+	public void testBean() {  
+	    Cat cat = new Cat("Mimi");  
+	  
+	    assertThat(cat, hasProperty("name", equalTo("Mimi")));  
+	}
+}
+```
+
+### equalToIgnoringCase
+It checks if two strings match regardless of capital or small letters.
+
+```java
+public class CatTest {  
+	@Test  
+	public void testStringEquality() {  
+	    String catNameInCaps = "RACHEL";  
+	  
+	    assertThat(catNameInCaps, equalToIgnoringCase("rachel"));  
+	}
+}
+```
+
+### containsString
+It checks if a string is contained in another string.
+
+```java
+public class CatTest {  
+	@Test  
+	public void testStringContains() {  
+	    String catName = "Joey The Cute";  
+	  
+	    assertThat(catName, containsString("Cute"));  
+	}
+}
+```
+
+## Assumptions
+We can think of assumptions like preconditions that have to be met for the test to pass. Let's check some examples.
```

Comment:
This section and the example don't really explain what assumptions are and why I should use them? Why not use an `assert` instead of an `assume`? What's the value?